clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2789,14 +2825,11 @@ void SYCLIntegrationHeader::startKernel(StringRef KernelName,
 }
 
 void SYCLIntegrationHeader::addParamDesc(kernel_param_kind_t Kind, int Info,
-                                         unsigned Offset) {
+                                         unsigned Offset,
+                                         unsigned NumOpenCLParams) {
   auto *K = getCurKernelDesc();
   assert(K && "no kernels");
-  K->Params.push_back(KernelParamDesc());
-  KernelParamDesc &PD = K->Params.back();
-  PD.Kind = Kind;
-  PD.Info = Info;
-  PD.Offset = Offset;
+  K->Params.push_back({Kind, Info, Offset, NumOpenCLParams});
```

Comment:
`.emplace_back()`?

Reply:
@keryell : emplace_back doesn't support aggregate initialization until C++20, and our codebase is C++14.

Reply:
> @keryell : emplace_back doesn't support aggregate initialization until C++20, and our codebase is C++14.

Then you know what you have to do. Please replace all the `-std=c++14` with `-std=c++20` ;-)
This is depressing all this time wasted writing old code... :-(

Reply:
> 
> 
> > @keryell : emplace_back doesn't support aggregate initialization until C++20, and our codebase is C++14.
> 
> Then you know what you have to do. Please replace all the `-std=c++14` with `-std=c++20` ;-)
> This is depressing all this time wasted writing old code... :-(

Wish I could!  It was hard enough to get the LLVM project switched over from a terrible-subset-of C++11 to C++14.